src/firewall.py

```diff
@@ -19,510 +19,414 @@
     from yunohost.service import _run_service_command
 
-    reloaded = False
-    errors = False
+    firewall = YunoFirewall()
 
     # Check if SSH port is allowed
-    ssh_port = _get_ssh_port()
-    if ssh_port not in firewall_list()["opened_ports"]:
-        firewall_allow("TCP", ssh_port, no_reload=True)
-
-    # Retrieve firewall rules and UPnP status
-    firewall = firewall_list(raw=True)
-    upnp = firewall_upnp()["enabled"] if not skip_upnp else False
-
-    # IPv4
-    try:
-        process.check_output("iptables -n -w -L")
-    except process.CalledProcessError as e:
-        logger.debug(
-            "iptables seems to be not available, it outputs:\n%s",
-            e.output.decode().strip(),
-        )
-        logger.warning(m18n.n("iptables_unavailable"))
-    else:
-        rules = [
-            "iptables -w -F",
-            "iptables -w -X",
-            "iptables -w -A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT",
-        ]
-        # Iterate over ports and add rule
-        for protocol in ["TCP", "UDP"]:
-            for port in firewall["ipv4"][protocol]:
-                rules.append(
-                    "iptables -w -A INPUT -p %s --dport %s -j ACCEPT"
-                    % (protocol, process.quote(str(port)))
-                )
-        rules += [
-            "iptables -w -A INPUT -i lo -j ACCEPT",
-            "iptables -w -A INPUT -p icmp -j ACCEPT",
-            "iptables -w -P INPUT DROP",
-        ]
-
-        # Execute each rule
-        if process.run_commands(rules, callback=_on_rule_command_error):
-            errors = True
-        reloaded = True
-
-    # IPv6
-    try:
-        process.check_output("ip6tables -n -L")
-    except process.CalledProcessError as e:
-        logger.debug(
-            "ip6tables seems to be not available, it outputs:\n%s",
-            e.output.decode().strip(),
-        )
-        logger.warning(m18n.n("ip6tables_unavailable"))
-    else:
-        rules = [
-            "ip6tables -w -F",
-            "ip6tables -w -X",
-            "ip6tables -w -A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT",
-        ]
-        # Iterate over ports and add rule
-        for protocol in ["TCP", "UDP"]:
-            for port in firewall["ipv6"][protocol]:
-                rules.append(
-                    "ip6tables -w -A INPUT -p %s --dport %s -j ACCEPT"
-                    % (protocol, process.quote(str(port)))
-                )
-        rules += [
-            "ip6tables -w -A INPUT -i lo -j ACCEPT",
-            "ip6tables -w -A INPUT -p icmpv6 -j ACCEPT",
-            "ip6tables -w -P INPUT DROP",
-        ]
-
-        # Execute each rule
-        if process.run_commands(rules, callback=_on_rule_command_error):
-            errors = True
-        reloaded = True
-
-    if not reloaded:
-        raise YunohostError("firewall_reload_failed")
-
-    hook_callback(
-        "post_iptable_rules", args=[upnp, os.path.exists("/proc/net/if_inet6")]
```

Comment:
Those should be adapted, and just dropped via `ynh_add_config --template nftable_rules --destination /etc/nftables.d/$app.conf` and then yunohost firewall reload.

Reply:
Although hmmmm I can see the very *hroumpf* extended logic implemented here, so I guess we can keep this hook based execution.

Reply:
(i wasnt pushing to keep the hook, dropping an additional conf file might be fine, i'm not sure, didn't really re-read the full logic)

Reply:
Yeah the logic in the file could be replaced by a template file, because the "source" of the logic (`/etc/openvpn/client.conf`) is generated in the `config` script.
But anyways that would be fine to keep the hook for some weird cases, although I would prefer it to disappear (it could be replaced by systemd PostExec to be run after nftables.service)

Reply:
This is a good reason for *not* wanting hooks : https://wiki.nftables.org/wiki-nftables/index.php/Atomic_rule_replacement#Warning_about_Shell_scripting_+_nftables

This is a good reason FOR wanting hooks : https://github.com/YunoHost-Apps/vpnclient_ynh/blob/179f9fa3354a3b6d2425653af0b6749e1be82f43/conf/hook_post-iptable-rules#L24

Reply:
The only apps relying on those hooks are :
* monitorix
* vpnclient
* incus

The apps directly calling iptables:
* hotspot
* vpnclient
* wireguard


Considering:
* incus can be easily patched
* wireguard is KO anyways


we only need to handle vpnclient and monitorix.

Here's monitorix post_iptables hook:
```
 1 #/bin/bash                                                                                                                                                                                                                    
 2                                                                                                                                                                                                                               
 3 if systemctl is-active __APP__.service  --quiet; then                                                                                                                                                                         
 4     systemctl stop monitorix.service                                                                                                                                                                                          
 5     sleep 1                                                                                                                                                                                                                   
 6     pkill -f "monitorix-httpd listening on" || true                                                                                                                                                                           
 7     systemctl start __APP__.service                                                                                                                                                                                           
 8 fi                                                                                                                                                                                                                            
 9                                                                                                                                                                                                                               
```
-> WTF ??
-> That can be handled as a systemd service that is started after nftables.

So the only app that will be problematic is vpnclient. We can patch it before yunohost is released with this branch merged, with a "if nftables ... if iptables ..."

Reply:
Ok things are simpler now. `nftables.service` will call `yunohost hooks callback post_iptable_rules` and `yunohost hooks callback post_nftables` and current apps will work (to be tested : iptables-compat)

Reply:
OK, reverted this, we now run the yunohost post_iptable hook after restarting nftables.